src/swarm.js

```diff
@@ -20,7 +20,7 @@ function Swarm () {
 
   self.port = parseInt(process.env.IPFS_SWARM_PORT, 10) || 4001
   self.connections = {} // {peerIdB58: {conn: <>, socket: <>}
-  self.handles = []
+  self.handles = {}
```

Comment:
No perf reason but the lookup you were trying to do if there was alread a handler wasn't working, so I changed it to an object, as otherwise we would have to search the whole list to check if we already have a handler.